### PR TITLE
Change Shibboleth to use SSL proxy in front of HTTP services

### PR DESCRIPTION
**Pull request by lower29**
Previously we were doing both SSL and Shibboleth in the same layer, by overriding the nginx service implementation and replacing that nginx container with one with Shibboleth and SSL enabled. However, this doesn't fit with how Jisc want to deploy it: they need the SSL layer, and the Shibboleth layer, to be seperate from the application layer.

This pull request replaces the current am-shib `nginx` service with two new services: `nginx-ssl` and `shib-sp-proxy`. The `nginx-ssl` service is just vanilla nginx with SSL certificates, the `shib-sp-proxy` service uses our [rdss-archivematica-shib-sp-proxy](https://github.com/JiscRDSS/rdss-archivematica-shib-sp-proxy/) image to provide the necessary Shibboleth SPs for our Dashboard and Storage Service applications.

This all sounds simple and straightforward but it's not; nothing ever is with Shibboleth!

The changes for this pull request build on those in #179, so that PR needs merging before this one. It also needs https://github.com/JiscRDSS/rdss-archivematica-shib-sp-proxy/pull/1 to be merged, too.

### CODE REVIEW

File: publish-images-playbook.yml [hunk 1]
```diff
@@ -69,6 +69,15 @@
             path: "./src/rdss-arkivum-nextcloud/"
             dockerfile: "Dockerfile"
 
+      - name: "RDSS Archivematica Shibboleth SP Proxy"
+        repo: "https://github.com/JiscRDSS/rdss-archivematica-shib-sp-proxy.git"
+        version: "dev/archivematica"
```
Comment (lower29) [inline publish-images-playbook.yml]: This should be changed to a semver version tag once the initial changes have been merged and we can create a tag.

**Comment by lower29** [comment]
Changes for https://github.com/JiscRDSS/rdss-archivematica-shib-sp-proxy/pull/1 have now been approved and merged.

File: publish-images-playbook.yml (hunk 1 above)
Comment (sevein) [reply publish-images-playbook.yml]: If you see that the branch is missing sorry that was me, I thought it wasn't needed anymore but then I realized it's still referenced from here even though the PR was merged! Sorry :)

Comment (lower29) [reply publish-images-playbook.yml]: Don't worry, I was only using `dev/archivematica` as a placeholder whilst developing. The required changes have all been merged now, so I've created a semver release (`v0.1.0`) and am using that instead.

**Pull request merged**